clients/manifest.go

```diff
@@ -183,6 +183,55 @@ func GenerateAndUploadManifests(sourceManifest m3u8.MediaPlaylist, targetOSURL s
 	return res, nil
 }
 
+// Generate a Master manifest for the FMP4 media playlists then write them to storage
+func GenerateAndUploadFragMp4Manifests(targetOSURL string, fmp4Manifests map[string]string, transcodedStats []*video.RenditionStats) error {
+
+	masterPlaylist := m3u8.NewMasterPlaylist()
+
+	//sort transcoded Stats and loop in order.
+	sort.Slice(transcodedStats, func(a, b int) bool {
+		if transcodedStats[a].BitsPerSecond > transcodedStats[b].BitsPerSecond {
+			return true
+		} else if transcodedStats[a].BitsPerSecond < transcodedStats[b].BitsPerSecond {
+			return false
+		} else {
+			var resolutionA = transcodedStats[a].Width * transcodedStats[a].Height
+			var resolutionB = transcodedStats[b].Width * transcodedStats[b].Height
+
+			return resolutionA > resolutionB
+		}
+	})
+
+	// For each profile, add a new entry to the master manifest
+	for i, profile := range transcodedStats {
```

Comment:
Do we already have this logic for standard HLS that we could reuse?

Reply:
Yeah i just copied for now. i'll turn this into a function.